src/lib/engine/utils.js

```diff
@@ -39,3 +39,11 @@ export function makeURLsMarkdown(text) {
 
   return transformedText
 }
+
+export const indicesOf = (list, predFn) => list.map((e, i) => predFn(e) ? i : null).filter(e => _.isFinite(e))
+export function getSubMatches(str, regex, matchIndex) {
```

Comment:
Agreed, but I think that belongs in a separate story. The changes to this file in this PR are rather minimal.